### PR TITLE
  insights: refactor capture group methods for readability and performance

### PR DESCRIPTION
**Pull request by DaivikDave**
Fixes #33148 
## Test plan
Tested by running go test and test cases under enterprise/internal/insights/query/ are passing

@sourcegraph/code-insights

### CODE REVIEW

File: enterprise/internal/insights/query/compute_graphql_results.go [hunk 1]
```diff
@@ -110,8 +110,10 @@ func (c ComputeMatchContext) RepoID() string {
 
 func (c ComputeMatchContext) Counts() map[string]int {
 	distinct := make(map[string]int)
-	for _, value := range c.MatchValues() {
```
Comment (leonore) [inline enterprise/internal/insights/query/compute_graphql_results.go:113]: this is the only function that consumes this, so we can delete the `MatchValues` helper I believe

**Review by leonore — COMMENTED** [review]
could

File: enterprise/internal/insights/query/compute_graphql_results.go (hunk 1 above)
Comment (DaivikDave) [reply enterprise/internal/insights/query/compute_graphql_results.go:113]: Yes, it's not used anywhere outside of compute_graphql_results . I have removed the helper in the latest commit.

**Review by leonore — APPROVED** [review]
the change makes sense to me. I tested it quickly locally by making a capture group insight and didn't see a difference in behaviour

**Comment by leonore** [comment]
hey @DaivikDave sorry this never got through to merging, I was never able to get the tests to run and got taken up by other things. this code was part of a path that was deprecated and is removed in #38416! sorry about this, really appreciate the interest you took in improving this

**Pull request closed without merging**